src/librustc_mir/transform/check_unsafety.rs

```diff
@@ -88,6 +91,21 @@ impl<'a, 'tcx> Visitor<'tcx> for UnsafetyChecker<'a, 'tcx> {
                         UnsafetyViolationKind::GeneralAndConstFn,
                     )
                 }
+
+                if let ty::FnDef(func_id, _) = func_ty.kind {
+                    let callee_features = self.tcx.codegen_fn_attrs(func_id).target_features;
+                    if !callee_features.is_empty() {
+                        let self_features =
+                            self.tcx.codegen_fn_attrs(self.body_did).target_features;
```

Comment:
Consider adding a method like `x_codgen_attrs.is_target_features_subset(y_codegen_attrs)` to express the intent in a less low level fashion. (Or if you don't think that will be reused, leave a comment here.)